### PR TITLE
added default behavior if no streams and epoch_size specified

### PR DESCRIPTION
**Pull request by snarayan21**
## Description of changes:
Added default behavior for when `epoch_size` is specified but no streams given in `StreamingDataset` instantiation.

Issue was that specifying `epoch_size` without any streams triggered a `ValueError` in the `apply_weights` method of the `Stream` class because it didn't want choose to be specified for the whole epoch unless weighting relatively. But with just one stream, the epoch_size would only apply to one stream, so we would choose `epoch_size` elements over the single dataset per epoch.

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [ ] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [x] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [x] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#submitting-a-contribution))
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: streaming/base/dataset.py [hunk 1]
```diff
@@ -306,6 +313,22 @@ def __init__(self,
             }
             for stream in streams:
                 stream.apply_default(default)
+        elif epoch_size_value:
+            # if there are no streams provided but the epoch_size is speficied
```
Comment (karan6181) [inline streaming/base/dataset.py:317]: nit: `speficied` -> `specified`. Also check the next line.

Comment (snarayan21) [reply streaming/base/dataset.py:317]: rare typo when made twice in a row 😂

Comment (snarayan21) [reply streaming/base/dataset.py:317]: modifying the `elif` case on this as discussed earlier

File: setup.py [hunk 2]
```diff
@@ -64,19 +64,10 @@
...
-    'toml==0.10.2',
-    'yamllint==1.32.0',
-    'moto>=4.0,<5',
-    'fastapi==0.100.0',
-    'pydantic==1.10.11',
-    'uvicorn==0.23.1',
+    'datasets>=2.4.0,<3', 'docformatter>=1.4', 'jupyter==1.0.0', 'pre-commit>=2.18.1,<4',
+    'pytest==7.4.0', 'pytest_codeblocks==0.16.1', 'pytest-cov>=4,<5', 'toml==0.10.2',
+    'yamllint==1.32.0', 'moto>=4.0,<5', 'fastapi==0.100.0', 'pydantic==1.10.11', 'uvicorn==0.23.1',
+    'pytest-split==0.8.1'
```
[the remark this answers is not in the document]
Comment (snarayan21) [reply setup.py:70]: will fix rn

**Review by karan6181 — APPROVED** [review]
LGTM. Thank You!

**Pull request merged**